source/client/HandView.py

```diff
@@ -85,300 +82,94 @@ def update(self):
-            return updated_wrapped_hand
-
-    def refreshXY(self, original, layout_option=1):
-        self.refresh_flag = False
-        """After sorting or melding, may wish to refresh card's xy coordinates """
-        maxX = UIC.Disp_Width - (self.hand_scaling[1] / 2)
-        if not layout_option == 1:
-            print('the only layout supported now is cards in a line, left to right')
-        refreshed = []
-        card_xy = (10, UIC.Table_Hand_Border + 40)
-        for element in original:
-            element.img_clickable.x = card_xy[0]
-            element.img_clickable.y = card_xy[1]
-            card_xy = (card_xy[0] + self.hand_scaling[1], card_xy[1])
-            refreshed.append(element)
-        if (card_xy[0] > maxX):
-            scalingfactor = maxX / card_xy[0]
-            self.hand_scaling = (scalingfactor * self.hand_scaling[0], scalingfactor * self.hand_scaling[1])
-            refreshed = self.rescaleCards(refreshed, self.hand_scaling[0])
-        return refreshed
-
-    def rescaleCards(self, original, card_scaling):
-        rescaled = []
-        for element in original:
-            loc_xy = (element.img_clickable.x, element.img_clickable.y)
-            scaledelement = UICardWrapper(element.card, loc_xy, card_scaling)
-            scaledelement.status = element.status
-            scaledelement.img_clickable.outline_index = element.img_clickable.outline_index
-            rescaled.append(scaledelement)
-        self.refresh_flag = True
-        return rescaled
-
-    def showHolding(self, wrapped_cards):
-        wrapped_cards.sort(key=lambda wc: wc.img_clickable.x)
-        for wrapped_element in wrapped_cards:
-            color = UIC.outline_colors[wrapped_element.img_clickable.outline_index]
-            loc_xy = (wrapped_element.img_clickable.x, wrapped_element.img_clickable.y)
-            wrapped_element.img_clickable.draw(self.display, loc_xy, color)
+            if self.event.type == pygame.MOUSEBUTTONDOWN:
+                HandAndFootButtons.ClickedButton(self, pos)
+                # ---  make below a generic helper function 'CardSelection' ----
+                for element in self.hand_info:
+                    # cannot select prepared cards, so not included in logic below.
+                    if element.img_clickable.isOver(pos):
+                        if element.status == 1:
+                            element.status = 0
+                            element.img_clickable.changeOutline(0)
+                        elif element.status == 0:
+                            element.status = 1
+                            element.img_clickable.changeOutline(2)
+
+            elif self.event.type == pygame.MOUSEMOTION:
+                HandAndFootButtons.MouseHiLight(self, pos)
+                HandManagement.MouseHiLight(self, pos)
+            elif self.event.type == pygame.KEYDOWN and self.num_wilds > 0:
+                self.assignWilds()
+
+    def assignWilds(self):
```

Comment:
Not required!
This can be a HandManagement.ManuallyAssign and you can do key events relating to the valid key options that were provided. It would need to be extended if/when other key options that included runs/suits were added, but that addition is such a big project already that I think that is fine.

Reply:
Moved existing method to HandManagement.manuallyAssign() with note that it currently only supports set-based games like HandAndFoot.